tests/draft2020-12/ref.json

```diff
@@ -432,5 +432,109 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "refs with relative uris and defs",
+        "schema": {
+            "$id": "http://example.com/schema1.json",
```

Comment:
I notice that you use the same `$id` for each test. Technically, that shouldn't matter because each test _should_ be run in isolation, but I've noticed that there has been an effort in the past to use unique identifiers. It's fine with me, but we should make sure everyone is ok with it if we are going to allow duplicate `$id`s across tests.

Reply:
Please use different $ids for each schema.  I *thought* we had a CI check in place for that, but maybe we just talked about it and no one has written it yet :)